### PR TITLE
Added VCA with 1 CV modulation source, and mono output.

### PR DESCRIPTION
**Pull request by LasseMunk**
Added VCA with 1 CV input and 1 mono output

### CODE REVIEW

File: Source/ProcessorFactory.h [hunk 1]
```diff
@@ -67,7 +69,9 @@ namespace synthvr
             SequenceProcessor = 18,
             DualVCAProcessor = 19,
             MatrixMixerProcessor = 20,
-            PrimeTimeProcessor = 21
+            PrimeTimeProcessor = 21,
+            Lasses_Sweet_Test_Module = 22,
+            BitRedux = 23
```
Comment (danielrothmann) [inline Source/ProcessorFactory.h]: Looks like a mixup with the two branches here - this can be removed

File: Source/ProcessorFactory.h [hunk 2]
```diff
@@ -142,6 +146,12 @@ namespace synthvr
                 case ProcessorID::PrimeTimeProcessor:
                     return reinterpret_cast<BaseProcessor*>(new class PrimeTimeProcessor());
 
+                case ProcessorID::Lasses_Sweet_Test_Module:
+                    return reinterpret_cast<BaseProcessor*>(new class Lasses_Sweet_Test_Module());
+                
+                case ProcessorID::BitRedux:
```
Comment (danielrothmann) [inline Source/ProcessorFactory.h]: This can also be removed on this branch

File: Source/Lasses_Sweet_Test_Module.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        AudioParameterFloat* gainParam;
+
+        AudioParameterFloat* gainCVAmountParam;
+
+        SmoothedValue<float> smoothedGainCVAmountParam;
+        float paramSmoothTime;
+
+        float normalizedGainCVModInput;
+
+        enum
```
Comment (danielrothmann) [inline Source/Lasses_Sweet_Test_Module.h]: Well done using this - makes the dsp code easier to read

File: Source/Lasses_Sweet_Test_Module.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "JuceHeader.h"
+#include "BaseProcessor.h"
+#include "ParameterUtils.h"
+
+using namespace juce;
+
+namespace synthvr
+{
+    class Lasses_Sweet_Test_Module : public BaseProcessor {
```
Comment (danielrothmann) [inline Source/Lasses_Sweet_Test_Module.h]: Let's change this to a more descriptive name such as SingleChannelVCA or something like that

File: Source/Lasses_Sweet_Test_Module.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        
+
+        if (!isInputConnected[gainCVModInput])
+        {
+            for (int sample = 0; sample < buffer.getNumSamples(); sample++) {
+
+                buffer.setSample(
+                    outputChannel, 
+                    sample, 
+                    gain.processSample(buffer.getSample(inputChannel, sample))
```
Comment (danielrothmann) [inline Source/Lasses_Sweet_Test_Module.cpp]: Does the gain not need smoothing when it's set using parameter only? Might be worth testing that scenario out on some low freq source audio

**Comment by danielrothmann** [comment]
Great start! I've put a few change suggestions in comments. When you have a look, you can commit your changes with message "Code review: [...your changes...]"

**Review by danielrothmann — CHANGES_REQUESTED** [review]

File: Source/Lasses_Sweet_Test_Module.cpp (hunk 5 above)
Comment (LasseMunk) [reply Source/Lasses_Sweet_Test_Module.cpp]: I might be wrong - but isn't it what the gain.setGainLinear and gain.processSample is doing ? Making the linear interpolation? :)

I have quite crap headphones right here, but I don't hear clicks manually adjusting the gain with input around 100 hz (ish) sinewave :)

Comment (danielrothmann) [reply Source/Lasses_Sweet_Test_Module.cpp]: Oh yes you are right, my bad. The JUCE dsp::gain component does smoothing 🤪

**Comment by LasseMunk** [comment]
Changes comitted and pushed :)

File: SynthVR-Native.jucer [hunk 6]
```diff
@@ -42,6 +42,16 @@
               file="Source/ProcessorAPIUtilities.h"/>
       </GROUP>
       <GROUP id="{2C78AE7D-EE4E-0F9F-C542-1D2EBDBBAAE8}" name="Processors">
+        <GROUP id="{35DE8CB3-7CE6-7E13-8130-E072DE0E31B7}" name="BitRedux">
```
Comment (danielrothmann) [inline SynthVR-Native.jucer]: Let's remove the bitredux from the producer config also

File: SynthVR-Native.jucer [hunk 7]
```diff
@@ -42,6 +42,16 @@
...
+          <FILE id="Kv8VCR" name="BitRedux.cpp" compile="1" resource="0" file="Source/BitRedux.cpp"/>
+          <FILE id="yK74i1" name="BitRedux.h" compile="0" resource="0" file="Source/BitRedux.h"/>
+        </GROUP>
+        <GROUP id="{BE696C9F-6FC6-0F49-7291-9511FE8544AD}" name="Lasses Sweet Modules">
```
Comment (danielrothmann) [inline SynthVR-Native.jucer]: Maybe we should name the group Utilities or something like that?

**Review by danielrothmann — CHANGES_REQUESTED** [review]

**Comment by LasseMunk** [comment]
reorganized Projuicer project :)

**Review by danielrothmann — APPROVED** [review]
Looks good!

**Pull request merged**